SCClassLibrary/Common/Collections/Event.sc

```diff
@@ -339,8 +339,19 @@ Event : Environment {
 						schedBundle.value(lag, dur + ~timingOffset, server,
 							[15 /* \n_set */, msg[2], \gate, 0])
 					}
+				},
+				// when making s_new msg, check if blockSize or newEx new msg
+				reblock: {|msg, label|
+					var bls = ~blockSize.value;
+					var hasBls = bls.notNil and: {~server.checkBlockSize(bls, label) };
+					var ups = ~upsample.value;
+					var hasUps = ups.notNil and: { ~server.checkUpsample(ups, label) };
+					if (hasBls or: hasUps) {
+						// "reblock! use 66. blockSize: % upsample: % \n".postf(bs, ups);
+						msg = msg.put(0, 66).insert(5, bls ? 0).insert(6, ups ? 0);
```

Comment:
Will `reblock` be the official name for all thses things? If yes, that's fine, if not, a more explicit name would help readers. Without knowing what it is about, you'd understand it as something that has to do with "blocking" the message.

Reply:
reblock is just a first quick name choice. The long name would have been ```convertToNewExMessageIfNeeded```, which seemed excessive. reblock is maybe too short. 
I was considering another option: a ```~newMsgHead``` function which creates 
the head part of a newMsg either for normal or reblock mode, 
but that would have been more code refactoring of all the message head code sections.
So ~reblock as now seemed safer.

Reply:
>  I was considering another option: a `~newMsgHead` function which creates the head part of a newMsg either for normal or reblock mode, but that would have been more code refactoring of all the message head code sections. So ~reblock by conversion as now seemed safer.

Yes, I went through the same consideration.

Reply:
It's a good first name. We can see – a shorter one would be `adjustBlockSize` (we don't have a general concept for both block size and oversampling, which could also be like `timeSpec`, or, well `reblock`)

Reply:
yes :-) so we stick with reblock for now - if we find a better name, we rename it.